### PR TITLE
Separate api loading from app screens

### LINKED ISSUE #77: Stuck on loading screen when api connection fails
currently, the user would get stuck on a loading page in case the selected API connection fails. also, there is no way for the user to change the selected server or go back to the previous state.

### proposed solution:
- add a dedicated screen for showing the connection state and  switch servers (`ApiLoadingNavigator`)
- main app screens would not be available unless we have a successful connection
- for switching server simply navigate to `ApiLoadingNavigator`

result:
https://user-images.githubusercontent.com/1168983/120181264-1b1a3d80-c20d-11eb-9b7b-95c687ec4b22.mp4

### PR DESCRIPTION
**Pull request by ahoseinian**
issue #77 

This PR solves the problem of getting stuck on the loading screen after a connection error by giving the user the possibility to switch to other networks

there are two major changes
- I removed `withNetworkSelect` and replaced it with a page that you can navigate to
- I created the new page under the Atomic/pages folder. I don’t know if you are a fan of that. I can change them to reflect the current folder structure of course.

the idea with the atomic structure is as follows:
atomic/
 - atoms -> small components btns, icons, texts & etc..
 - molecules -> component containing couple of atoms
 - organisms -> component containing molecules & atoms
 - pages -> navigation screens contains everything
 
also, I would normally don't break anything down into smaller pieces unless it'll be used in more than just one place in the app otherwise it can stay on the page file.

demo:
https://user-images.githubusercontent.com/1168983/120181264-1b1a3d80-c20d-11eb-9b7b-95c687ec4b22.mp4

### CODE REVIEW

File: src/Atomic/pages/ApiLoadingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  /// navigate to app in case of successfully connecting
+  const prevApi = React.useRef(!!api);
+  React.useEffect(() => {
+    if (api && !prevApi.current) {
+      navigation.navigate('App');
+    }
+  }, [navigation, api]);
+
+  /////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (geastwood) [inline src/Atomic/pages/ApiLoadingPage.tsx:28]: what is this?

File: src/navigation/AppNavigator.tsx [hunk 2]
```diff
@@ -34,9 +37,12 @@ function DashboardStackNavigator() {
   );
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (geastwood) [inline src/navigation/AppNavigator.tsx:40]: let's remove this.

File: src/navigation/AppNavigator.tsx [hunk 3]
```diff
@@ -55,4 +61,54 @@ function AppNavigator() {
...
```
Comment (geastwood) [inline src/navigation/AppNavigator.tsx:64]: same here

File: src/navigation/AppNavigator.tsx [hunk 4]
```diff
@@ -55,4 +61,54 @@ function AppNavigator() {
...
+        <AppStack.Screen name={'App'} component={DrawerNavigator} />
+      ) : undefined}
+      <AppStack.Screen name={'ApiNavigator'} component={ApiLoadingNavigator} />
+    </AppStack.Navigator>
+  );
+}
+
 export default AppNavigator;
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (geastwood) [inline src/navigation/AppNavigator.tsx:84]: and here

**Review by geastwood — APPROVED** [review]
Small things to remove and mergeable.

File: src/Atomic/pages/ApiLoadingPage.tsx (hunk 1 above)
Comment (ahoseinian) [reply src/Atomic/pages/ApiLoadingPage.tsx:28]: I am used to separating stuff with a line of comment like this. I will remove them ✅

**Review by silva-fj — COMMENTED** [review]
@ahoseinian It looks good to me 👍🏼 

> I created the new page under the Atomic/pages folder. I don’t know if you are a fan of that. I can change them to reflect the current folder structure of course.

Let's not introduce a new folder structure as part of this PR. We can open an issue and have a discussion (there are still some open topics about the project we should discuss). cc @geastwood

**Comment by ahoseinian** [comment]
@franjs make sense I will refactor this to current format and then merge it

**Pull request merged**